core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1065,7 +1084,7 @@ private[spark] class BlockManager(
           try {
             replicate(blockId, bytesToReplicate, level, remoteClassTag)
           } finally {
-            bytesToReplicate.unmap()
+            bytesToReplicate.dispose()
```

Comment:
I'm afraid this may counteract the effort we made in https://github.com/apache/spark/pull/16499

Ideally `unmap` and `dispose` do different things

cc @mallman